core/toolbox.js

```diff
@@ -470,23 +533,23 @@ Blockly.Toolbox.prototype.updateColourFromTheme = function() {
 
 /**
  * Updates the background colour of the selected category.
- * @param {!Blockly.Toolbox.TreeNode} tree Starting point of tree.
+ * @param {!Blockly.tree.TreeNode} tree Starting point of tree.
  *     Defaults to the root node.
  * @private
  */
 Blockly.Toolbox.prototype.updateSelectedItemColour_ = function(tree) {
-  var selectedItem = tree.selectedItem_;
+  var selectedItem = tree.getSelectedItem();
   if (selectedItem) {
     var hexColour = selectedItem.hexColour || '#57e';
     selectedItem.getRowElement().style.backgroundColor = hexColour;
-    tree.toolbox_.addColour_(selectedItem);
+    tree.getToolbox().addColour_(selectedItem);
   }
 };
 
 
 /**
  * Recursively add colours to this toolbox.
- * @param {Blockly.Toolbox.TreeNode=} opt_tree Starting point of tree.
+ * @param {Blockly.tree.BaseNode=} opt_tree Starting point of tree.
```

Comment:
This is a `BaseNode` but on line 505 you have a `TreeNode` for the exact same description of the parameter.  What's the difference, and why is this one not a `TreeNode`?

Reply:
TreeControl and TreeNode both inherit from BaseNode. For methods that begin at the root node (TreeControl) and recurse through their children calling the same method (a common theme here), they should use BaseNode as either subclass could be the input.
I wonder why closure didn't complain about the other one, but yes the other one should also be updated to BaseNode.